packages/alfa-rules/test/sia-r8/rule.spec.tsx

```diff
@@ -0,0 +1,200 @@
+import { test } from "@siteimprove/alfa-test";
+
+import { Document } from "@siteimprove/alfa-dom";
+
+import R8, { Outcomes } from "../../src/sia-r8/rule";
+
+import { evaluate } from "../common/evaluate";
+import { passed, failed, inapplicable } from "../common/outcome";
+
+test("evaluate() passes an input element with accessible name", async (t) => {
+  const target = <input />;
+
+  const label = (
+    <label>
+      first name
+      {target}
+    </label>
+  );
+
+  const document = Document.of([label]);
+
+  t.deepEqual(await evaluate(R8, { document }), [
+    passed(R8, target, {
+      1: Outcomes.HasName,
+    }),
+  ]);
+});
+
+test("evaluate() passes an input element with aria label", async (t) => {
+  const target = <input aria-label="last name" disabled />;
+
+  const label = <div>last name</div>;
+
+  const document = Document.of([label, target]);
```

Comment:
That's a case where the `<div>` is added in ACT rule to provide "good" code (the `<input>` needs a visual label in addition to the accessible name), and we don't care as part of a test suite, so we can keep it small.
```suggestion
  const document = Document.of([target]);
```